scripts/stakingSimulator.js

```diff
@@ -0,0 +1,633 @@
+            await Helper.mineNewBlockAt(
+                currentBlockTime + Helper.getRandomInt(10, epochPeriod.toNumber() / 8)
+            );
+            break;
+        default:
+            console.log("unexpected operation: " + operation);
+            break;
+    }
+}
+
+module.exports.doFuzzStakeTests = async function(
+    kyberStaking, NUM_RUNS, kncToken, stakers, epochPeriod
+) {
+    let result;
+    let validity;
+    for(let loop = 0; loop < NUM_RUNS; loop++) {
+        let operation = StakeGenerator.genNextOp(loop, NUM_RUNS);
+        switch(operation) {
+            case DEPOSIT:
+                result = await StakeGenerator.genDeposit(kncToken, stakers);
+                result.dAddress = await kyberStaking.getLatestRepresentative(result.staker);
+                console.log(result.msg);
+                console.log(`Deposit: staker ${result.staker}, amount: ${result.amount}`);
+                validity = await executeAndVerifyDepositInvariants(kyberStaking, result, epochPeriod);
+                depositRuns = logResult(validity, depositRuns);
+                break;
+
+            case DELEGATE:
+                result = await StakeGenerator.genDelegate(stakers);
+                console.log(result.msg);
+                console.log(`Delegate: staker ${result.staker}, address: ${result.dAddress}`);
+                validity = await executeAndVerifyDelegateInvariants(kyberStaking, result, epochPeriod);
+                delegateRuns = logResult(validity, delegateRuns);
+                break;
+
+            case WITHDRAW:
+                result = await StakeGenerator.genWithdraw(kyberStaking, stakers);
+                result.dAddress = await kyberStaking.getLatestRepresentative(result.staker);
+                console.log(result.msg);
+                console.log(`Withdrawal: staker ${result.staker}, amount: ${result.amount}`);
+                executeWithdraw(kyberStaking, result, epochPeriod);
+                // validity = await verifyWithdrawInvariants(kyberStaking, stakers, result);
+                validity = {isValid: true};
+                withdrawRuns = logResult(validity, withdrawRuns);
+                break;
+            case NO_ACTION:
+                console.log("do nothing for this epoch...");
+                // Advance time by a bit
+                let currentBlockTime = await Helper.getCurrentBlockTime();
+                await Helper.mineNewBlockAt(
+                    currentBlockTime + Helper.getRandomInt(10, epochPeriod.toNumber() / 8)
+                );
+                // validity = await verifyNoActionInvariants(kyberStaking, stakers, result);
+                validity = {isValid: true};
+                noActionRuns = logResult(validity, noActionRuns);
+                break;
+            default:
+                console.log("unexpected operation: " + operation);
+                break;
+        }
+    }
+
+    console.log(`--- SIM RESULTS ---`);
```

Comment:
Please call this run results or fuzz results.
Please add print to number of operations performed (Number of runs)